### DISCUSSION TITLE
usb_bulk_write returned -110 should be 9

### DISCUSSION THREAD

**Discussion by 1458861693** [libusb/libusb#1405] [Q&A]
There is a Ketek SDD（silicon drift detector）in my laboratory，which data is read by a USB Device（electronic instrument）.
I have used the USB device in EPICS（experimental physic and industry Control system）on CentOS7. 
when I start the system, it appears error as the following picture. 
I think this error may be caused by libusb, can you give me some help to solve it? 
Looking forward to your reply, Thanks！
<img width="836" alt="图片1" src="https://github.com/libusb/libusb/assets/74239651/fb398c12-9a96-45eb-96e0-bc62b357b5d6">
developing environment：
Linux version：
Linux localhost,localdomain 3.10.0-1160.105.l.el7.x86 64 #1 SMP Thu Dec 7 15:39:45 UTC 2023 x86 64 x86 64 x86 64 GNU/Linux
libusb version：
libusb-devel-0.1.4-3el7.x8664

**Comment by mcuee** [comment]
Looks like you are using very old version of libusb-0.1 which is not supported by libusb project since long ago.
https://rpm.pbone.net/info_idpl_55098487_distro_redhatel7_com_libusb-devel-0.1.4-3.el7.i686.rpm.html

Please upgrade to latest version of libusb (1.0.26 release or libusb-1.0.27-rc1) and libusb-compat-0.1 (libusb-compat-0.1.8) and test again.

Last release of libusb-0.1 was in year 2006.
https://sourceforge.net/projects/libusb/files/libusb-0.1%20%28LEGACY%29/

As far as I know, only Debian/Ubuntu still supports libusb-0.1.
https://packages.debian.org/unstable/libs/libusb-0.1-4

**Comment by mcuee** [comment]
Basically you can try your luck with CentOS or Redhat if you want the support. I will close this issue.

**Comment by mcuee** [comment]
Reference:
* https://github.com/epics-modules/dxp/issues/10

**Comment by 1458861693** [comment]
Thanks for your reply very much.
This reference is also committed by me, but this problem is not solved yet.
Thanks for your help very much，I will try the suggested version of libusb on Ubuntu 20.04.
Thanks again.

**Comment by mcuee** [comment]
> Thanks for your reply very much. This reference is also committed by me, but this problem is not solved yet. Thanks for your help very much，I will try the suggested version of libusb on Ubuntu 20.04. Thanks again.

Yes you can try that. But again that libusb-0.1 legacy version is only supported by Debian/Ubuntu and not libusb project. If it still does not work, I will recommend using libusb-compat-0.1 (on top of libusb-1.0).

If you use newer version of CentOS or use Fedora, they will use libusb-compat-0.1 (on top of libusb-1.0).

**Comment by 1458861693** [comment]
I try the libusb-1.0.26 on CentOS7，but when I compile this package，it appears this error：
![image](https://github.com/libusb/libusb/assets/74239651/22cedf32-803b-4c0d-8cd1-f4786e3c0a52)

Is this mean the version of my gcc compiler is old？ Should I upgrade it？ 
I have tried libusb-0.1.10a，libusb-1.0.26 with libusb-compat-0.1.8 on Ubuntu20.04  and libusb-0.1.12, libusb-1.0.21 with libusb-compat-0.1.8 on CentOS7 respectively，but it also doesn't work.
I really wish you can give me more help to solve it.

**Comment by mcuee** [comment]
I will not be able to help you much on CentOS 7. 

But libusb 1.0.26 and libusb-compat-0.1.8 should work fine under Ubuntu 20.04. You should try them and post your log here. Basically you should build libusb-1.0.26 and libusb-compat-0.1.8 from source under Ubuntu 20.04. Then your should rebuild your application against libusb-compat-0.1 (and not the system provided legacy libusb-0.1) and then test again.

**Comment by 1458861693** [comment]
Thanks for your suggestions.
I build libusb-1.0.26 and libusb-compat-0.1.8 from source code under Ubuntu 20.04.
I will rebuild my application and post my log here.
Thank you very much. Good luck!
